target_chains/ethereum/contracts/contracts/pulse/Pulse.sol

```diff
@@ -174,7 +193,10 @@ abstract contract Pulse is IPulse, PulseState {
         uint256 callbackGasLimit
     ) public view override returns (uint128 feeAmount) {
         uint128 baseFee = _state.pythFeeInWei;
-        uint256 gasFee = callbackGasLimit * tx.gasprice;
+        uint128 providerFeeInWei = _state
+            .providers[_state.defaultProvider]
+            .feeInWei;
+        uint256 gasFee = callbackGasLimit * providerFeeInWei;
```

Comment:
alright so now provider fee is per gas, and pyth fee is the constant, right?

Reply:
yup exactly

`baseFee` is the fixed fee to Pyth
`providerFeeInWei` is the provider's per-gas fee